app/src/main/java/com/eveningoutpost/dexdrip/Models/blueReader.java

```diff
@@ -1,20 +1,41 @@
 package com.eveningoutpost.dexdrip.Models;
+
+import android.text.format.DateFormat;
 import com.eveningoutpost.dexdrip.Home;
 import java.nio.ByteBuffer;
 import java.util.Date;
 import com.eveningoutpost.dexdrip.Services.DexCollectionService;
 import com.eveningoutpost.dexdrip.Models.UserError.Log;
+import com.eveningoutpost.dexdrip.UtilityModels.Notifications;
 import com.eveningoutpost.dexdrip.UtilityModels.PersistentStore;
 import com.eveningoutpost.dexdrip.UtilityModels.Pref;
 import com.eveningoutpost.dexdrip.xdrip;
+import com.eveningoutpost.dexdrip.R;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.regex.*;
+
+import static com.eveningoutpost.dexdrip.utils.FileUtils.getExternalDir;
+import static com.eveningoutpost.dexdrip.utils.FileUtils.makeSureDirectoryExists;
+
 
 /**
  * Created by MasterPlexus on 11.12.2017.
  */
 
 public class blueReader {
     private static final String TAG = "blueReader";
+    private static final String BatLog="/BatteryLog.csv";
     private static int counterHibernated = 0;
+    private static Matcher tempVers;
+
+    private final static byte[] shutdown = new byte[]{0x6B};        // Char 'k'
+    private final static byte[] requestValue = new byte[]{0x6C};    // Char 'l'
+    private final static byte[] goHybernate = new byte[]{0x68};     // Char 'h'
+    private final static byte[] restart = new byte[]{0x79};         // Char 'y'
```

Comment:
Correct ordering is "private static final"

Reply:
Not seen that this makes a difference, will change it.